modules/inmobiBidAdapter.js

```diff
@@ -0,0 +1,344 @@
+        });
+      });
+      return urls;
+    }
+  },
+
+  /**
+   * Validates if a bid request contains the required parameters for InMobi.
+   *
+   * @param {Object} bid - Bid request to validate.
+   * @returns {boolean} True if the bid request is valid, otherwise false.
+   */
+  isBidRequestValid: (bid) => {
+    if (!(bid && bid.params && bid.params.plc)) {
+      return false;
+    }
+
+    return true;
+  },
+
+  /**
+   * Builds the server request from bid requests for InMobi.
+   *
+   * @param {BidRequest[]} bidRequests - Array of bid requests.
+   * @param {Object} bidderRequest - Additional request details.
+   * @returns {ServerRequest} The server request for bidding.
+   */
+  buildRequests: (bidRequests, bidderRequest) => {
+    const data = CONVERTER.toORTB({ bidderRequest, bidRequests });
+
+    if (data) {
+      const requestPayload = {
+        method: POST_METHOD,
+        url: BID_ENDPOINT,
+        data: data,
+        options: {
+          contentType: PLAIN_CONTENT_TYPE,
+          crossOrigin: true,
+          withCredentials: true
+        }
+      };
+      return requestPayload;
+    }
+  },
+
+  /**
+   * Interprets the server response and formats it into bids.
+   *
+   * @param {Object} response - Response from the server.
+   * @param {ServerRequest} request - Original bid request.
+   * @returns {Bid[]} Parsed bids or configurations.
+   */
+  interpretResponse: (response, request) => {
+    if (typeof response?.body == 'undefined') {
+      return [];
+    }
+
+    const interpretedResponse = CONVERTER.fromORTB({ response: response.body, request: request.data });
+    const bids = interpretedResponse.bids || [];
+
+    return bids;
+  },
+
+  /**
+   * Callback to report timeout event.
+   *
+   * @param {TimedOutBid[]} timeoutData - Array of timeout details.
+   */
+  onTimeout: (timeoutData) => {
+    report('onTimeout', timeoutData);
+  },
+
+  /**
+   * Callback to report targeting event.
+   *
+   * @param {Bid} bid - The bid object
+   */
+  onSetTargeting: (bid) => {
+    report('onSetTargeting', bid?.meta);
+  },
+
+  /**
+   * Callback to report successful ad render event.
+   *
+   * @param {Bid} bid - The bid that successfully rendered.
+   */
+  onAdRenderSucceeded: (bid) => {
+    report('onAdRenderSucceeded', bid?.meta);
+  },
+
+  /**
+   * Callback to report bidder error event.
+   *
+   * @param {Object} errorData - Details about the error.
+   */
+  onBidderError: (errorData) => {
+    report('onBidderError', errorData);
+  },
+
+  /**
+   * Callback to report bid won event.
+   *
+   * @param {Bid} bid - The bid that won the auction.
+   */
+  onBidWon: (bid) => {
+    report('onBidWon', bid?.meta);
+  }
+
+};
+
+function isReportingAllowed(loggingPercentage) {
+  return loggingPercentage != 0;
+}
+
+function report(type, data) {
+  if (!data) {
+    return;
+  }
+  if (['onBidWon', 'onAdRenderSucceeded', 'onSetTargeting'].includes(type) && !isReportingAllowed(data.loggingPercentage)) {
```

Comment:
where is loggingPercentage getting set?

Reply:
Hi @patmmccann ,`loggingPercentage` will be set by our backend in `response.seatbid.bid[i].ext.loggingPercentage`